src/matrixvariate.jl

```diff
@@ -215,21 +215,23 @@ function Distributions._rand!(rng::AbstractRNG, d::TuringInverseWishart, A::Abst
     A .= inv(cholesky!(X))
 end
 
-# TODO: Remove when available in Distributions
-for T in (:MatrixBeta, :MatrixNormal, :Wishart, :InverseWishart,
-          :TuringWishart, :TuringInverseWishart,
-          :VectorOfMultivariate, :MatrixOfUnivariate)
-    @eval begin
-        Distributions.loglikelihood(d::$T, X::AbstractMatrix{<:Real}) = logpdf(d, X)
-        function Distributions.loglikelihood(d::$T, X::AbstractArray{<:Real,3})
-            (size(X, 1), size(X, 2)) == size(d) || throw(DimensionMismatch("Inconsistent array dimensions."))
-            return sum(i -> _logpdf(d, view(X, :, :, i)), axes(X, 3))
-        end
-        function Distributions.loglikelihood(
-            d::$T,
-            X::AbstractArray{<:AbstractMatrix{<:Real}},
-        )
-            return sum(x -> logpdf(d, x), X)
+# Only needed in Distributions < 0.24
+if !DISTRIBUTIONS_HAS_GENERIC_UNIVARIATE_PDF
```

Comment:
Why is this not needed? Please provide an explanation and a benchmark to make sure there is no regression especially for ReverseDiff and Zygote.

Reply:
Because Distributions implements this now for any `MatrixDistribution` (as it should): https://github.com/JuliaStats/Distributions.jl/blob/ae2d6c5c68abc0aeafdc249ca60df058c19279cb/src/matrixvariates.jl#L225-L232

I don't think a benchmark is useful here since it is exactly the same implementation.

Reply:
Can we just make this a check of the Distributions version?

Reply:
AFAIK there's no official API or way to check the version of certain packages, that's why DISTRIBUTIONS_HAS_GENERIC_UNIVARIATE_PDF serves as a proxy (another example is https://github.com/TuringLang/Turing.jl/blob/beb440593b7a61f299a875abb84f3521fdbf3ada/src/Turing.jl#L61-L66).